app/ante/evm/aspect_context.go

```diff
@@ -41,14 +42,21 @@ func (aspd AspectRuntimeContextDecorator) AnteHandle(ctx cosmos.Context, tx cosm
 
 		// Aspect Runtime Context Lifecycle: create AspectRuntimeContext
 		// this handler is for eth transaction only, should be 1 message for eth transaction.
-		ethTxContext := types.NewEthTxContext(msgEthTx.AsEthCallTransaction())
+		evmConfig, err := aspd.evmKeeper.EVMConfigFromCtx(ctx)
+		if err != nil {
+			return ctx, fmt.Errorf("failed to get evm config from context: %w", err)
+		}
+		ethTxContext := types.NewEthTxContext(msgEthTx.AsEthCallTransaction()).WithEVMConfig(evmConfig)
+
 		aspectCtx := types.NewAspectRuntimeContext()
 		protocol := provider.NewAspectProtocolProvider(aspectCtx.EthTxContext)
 		jitManager := inherent.NewManager(protocol)
 
 		// update EthTxContext and JIT manager
+		aspectCtx.SetEthBlockContext(types.NewEthBlockContextFromHeight(ctx.BlockHeight()))
 		aspectCtx.SetEthTxContext(ethTxContext, jitManager)
 		aspectCtx.WithCosmosContext(ctx)
+		aspectCtx.CreateStateObject()
```

Comment:
Yes, we need it here because the Aspect State Object is based on the current cosmos context, since in `ApplyMessageWithConfig` the cosmos context might be replaced, so it is safer for us to create aspect state object both here and in the `ApplyMessageWithConfig`